crates/libcgroups/src/v1/freezer.rs

```diff
@@ -79,9 +81,15 @@ impl Freezer {
                         if i % 25 == 24 {
                             thread::sleep(time::Duration::from_millis(10));
                         }
-
-                        let r = Self::read_freezer_state(cgroup_root)?;
-                        match r.trim() {
+                        let mut state = String::new();
+                        OpenOptions::new()
+                            .create(false)
+                            .read(true)
+                            .open(cgroup_root.join(CGROUP_FREEZER_STATE))
+                            .wrap_open(cgroup_root)?
+                            .read_to_string(&mut state)
+                            .wrap_read(cgroup_root)?;
+                        match state.trim() {
```

Comment:
is it possible to re used the read_freezer_state here instead of doing a read again?

Reply:
I also thought about that. The logic here is simpler, the wait/retry are handled in outside for-loop. While in read_freezer_state, it keeps waiting if the state is freezing. If we want to reuse read_freezer_state, we need an additional flag to control whether to wait on freezing state. I feel like it's fine to have a little bit duplication here. The go implementation also doesn't reuse GetState method in Set method. https://github.com/opencontainers/cgroups/blob/6a793b6ea778415b3c0f7a3d2efdb70c786fad92/fs/freezer.go#L86-L97